trl/trainer/grpo_trainer.py

```diff
@@ -674,22 +721,79 @@ def _prepare_inputs(self, inputs: dict[str, Union[torch.Tensor, Any]]) -> dict[s
             inputs = self._generate_and_score_completions(inputs)
         return inputs
 
+    def _prepare_inputs_for_reward_module(
+        self,
+        *,
+        inputs: dict[str, Union[torch.Tensor, Any]],
+        reward_processing_class: PreTrainedTokenizerBase,
+        prompts: list[str],
+        completions: list[str],
+        images=None,
+    ) -> dict[str, torch.Tensor | Any]:
+        if is_conversational(inputs[0]):
+            messages = [{"messages": p + c} for p, c in zip(prompts, completions)]
+            texts = [apply_chat_template(x, reward_processing_class)["text"] for x in messages]
+        else:
+            texts = [p + c for p, c in zip(prompts, completions)]
+        if images is None:
+            reward_inputs = reward_processing_class(
+                text=texts,
+                return_tensors="pt",
+                padding=True,
+                padding_side="right",
+                add_special_tokens=False,
+            )
+        else:
+            reward_inputs = reward_processing_class(
+                images=images,
+                text=texts,
+                return_tensors="pt",
+                padding=True,
+                padding_side="right",
+                add_special_tokens=False,
+            )
+        reward_inputs = super()._prepare_inputs(reward_inputs)
+
+        return reward_inputs
+
     def _generate_and_score_completions(
         self, inputs: dict[str, Union[torch.Tensor, Any]]
     ) -> dict[str, Union[torch.Tensor, Any]]:
         device = self.accelerator.device
         prompts = [x["prompt"] for x in inputs]
+        images = [x["image"] for x in inputs if "image" in x]
```

Comment:
I noticed the DPO trainer already supports VLMs but expects the column to be called "images", as a list of PIL images:

https://github.com/huggingface/trl/blob/26d86757a7c7e24e397ea44f57ecce6031dfac01/trl/trainer/dpo_trainer.py#L642

I think this should be "images" too, for consistency, and to support the case of multiple images per prompt.

Also, note that for that case the vLLM server would have to be able to receive the `limit_mm_per_prompt` argument.